### DISCUSSION TITLE
Increasing the stops buttons

### DISCUSSION THREAD

**Discussion by Jean-Gabriel1982** [GrandOrgue/GoOdf#66] [General]
Hello,
Is it possible to increase the stops buttons ?

I've selected 2nd one and try to increase de width and the height from default (65) to 80 for example ... but the result is weird 😀 

Example the "Tirasse GO"
![image](https://github.com/larspalo/GOODF/assets/20370038/3053dc2c-2666-4b50-837c-eb16f8fb1b9e)

Many thanks !

**Comment by larspalo** [comment]
GrandOrgue itself doesn't scale the images when you increase their size by changing their width and height. Instead they are tiled if larger size is used or cropped if a smaller size is used. If you want larger sizes than the built in bitmaps have you basically should provide the (custom) bitmaps yourself or live with the way GO can scale up the whole panel when you increase the (window) size which will affect all elements on that panel.

It's fairly easy to open up the already existing stop on/off bitmaps (from https://github.com/GrandOrgue/grandorgue/tree/master/src/images) in an image editor (like GIMP) and just scale them up there and save them back under a new name as a custom stop bitmap that you then reference to. The results of an up scaling from 65 to 80 is not too bad:

![Stop01off80](https://github.com/larspalo/GOODF/assets/6173941/df1bbc04-fe83-486f-a3f9-c54a3f844097)
![Stop01on80](https://github.com/larspalo/GOODF/assets/6173941/71d56947-54b6-47e4-9e41-281cbea56a03)

vs the originals

![drawstop01off](https://github.com/larspalo/GOODF/assets/6173941/58720e8e-b5de-4087-a84f-08e361abddcb)
![drawstop01on](https://github.com/larspalo/GOODF/assets/6173941/f5311ddb-6eb9-47db-9cf0-5f1ec1652f4e)

**Reply by Jean-Gabriel1982** [reply]
Ok super ! Many thanks for the answer !
I think I will create custom bitmaps for my stops ..

I'm show you the result 😄

**Reply by Jean-Gabriel1982** [reply]
Hello,

So I've designed some new buttons
![image](https://github.com/larspalo/GOODF/assets/20370038/10af895e-59b4-4413-a254-726631203865)

Today I've a problem in creating a stop .. I have four directories 
- Front left
- Front right 
- Rear left
- Rear right

I create four ranks for each directory and then a stop in my Recit, and I attribue the 4 ranks to my stop (referenced ranks). 

The problem : 
![image](https://github.com/larspalo/GOODF/assets/20370038/4586f69e-a79c-4648-8e03-c1587f3cef62)

Do you have an idea ?
Many thanks !

**Reply by larspalo** [reply]
I would suspect that the sample set you're using samples from (a Sonus Paradisi one) has a "trem" folder in each of the sample directories. However, this folder doesn't contain any real samples but instead files used for the synthetic tremulant of HW. So, in short - you need to remove the "trem" text from the "Tremulant folder prefix:" textfield so that GOODF won't try to load the files as samples - which GO won't like. See image below for some hints.

![CommonSPloadingOptions](https://github.com/larspalo/GOODF/assets/6173941/125b625e-e7c6-45bd-afaf-181c6a835df4)

You have two options for fixing the existing ranks.

1. Clear all the pipes for the rank and then set the correct loading options and re-load the pipes.
2. For each pipe you'd have to remove the tremulant samples that shouldn't be there. This will take much longer time than the first option.

**Reply by Jean-Gabriel1982** [reply]
Many thanks it works !
Except if I add the folder "trem" with the button "add trémulant samples from.." if I add that I have the same errors. 
If not everything works. 

Regarding the option "load release in attack", I just right click on the first pipe "attack" and then put the load option to "no" and copy to the same directory...
It works I hope this is the right way ...

**Reply by larspalo** [reply]
> Except if I add the folder "trem" with the button "add trémulant samples from.." if I add that I have the same errors

Yes, if you add the same synthetic tremulant information files with the "Add tremulant samples from..." button you'll get the same errors. You should only use that button if you know where any real tremulant samples exist for the pipes. If such tremulant samples exist in the SP sample sets there will usually be in separate directories for each division with a _TR or similar addition to the name - but there's no real standard so you'll have to look for it in the specific sample set you have.

> I just right click on the first pipe "attack" and then put the load option to "no" and copy to the same directory...

Sure, that works too!
